### PR TITLE
prevent no content helper on delete list records

### PR DESCRIPTION
**Pull request by msh-odoo**
Purpose
Change the pager if there are still remaining records after the deletion of a record.

SPEC
When there is only one record on the last page of a list view and we delete it, the noContentHelper shows.

TASK 2446911

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/js/views/basic/basic_model.js [hunk 1]
```diff
@@ -448,6 +448,12 @@ var BasicModel = AbstractModel.extend({
                     if (parent && parent.type === 'list') {
                         parent.data = _.without(parent.data, record.id);
                         delete self.localData[record.id];
+                        // Check if we are on last page and all record are deleted from current
```
Comment (SimonGenin) [inline addons/web/static/src/js/views/basic/basic_model.js:451]: records*

File: addons/web/static/src/js/views/basic/basic_model.js [hunk 2]
```diff
@@ -448,6 +448,12 @@ var BasicModel = AbstractModel.extend({
...
+                        // page i.e. if there is no state.data.length then go to previous page
+                        // if (!parent.data.length && parent.count <= (parent.offset + parent.limit)) {
```
Comment (SimonGenin) [inline addons/web/static/src/js/views/basic/basic_model.js:453]: forgotten code as comment ?

**Review by SimonGenin — CHANGES_REQUESTED** [review]

**Comment by msh-odoo** [comment]
@SimonGenin Changes done, Thanks.

File: addons/web/static/src/js/views/basic/basic_model.js [hunk 3]
```diff
@@ -448,6 +448,11 @@ var BasicModel = AbstractModel.extend({
                     if (parent && parent.type === 'list') {
                         parent.data = _.without(parent.data, record.id);
                         delete self.localData[record.id];
+                        // Check if we are on last page and all records are deleted from current
+                        // page i.e. if there is no state.data.length then go to previous page
+                        if (!parent.data.length && parent.offset >= parent.limit) {
+                            parent.offset = (parent.offset - parent.limit) > 0 ? (parent.offset - parent.limit) : 0;
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/basic/basic_model.js:454]:
```suggestion
                        if (!parent.data.length && parent.offset > 0) {
                            parent.offset = Math.max(parent.offset - parent.limit, 0);
```

File: addons/web/static/src/js/views/basic/basic_model.js [hunk 4]
```diff
@@ -1290,6 +1295,15 @@ var BasicModel = AbstractModel.extend({
                 // optionally clear the DataManager's cache
                 self._invalidateCache(parent);
                 return self.reload(parentID);
+            }).then(function (datapoint) {
+                // if there are no records to display and we are not on first page(we check it
+                // by checking offset is greater than limit i.e. we are not on first page)
+                // reason for adding logic after reload to make sure there is no records after operation
+                if (parent && parent.type === 'list' && !parent.data.length && parent.offset >= parent.limit) {
+                    parent.offset = (parent.offset - parent.limit) > 0 ? (parent.offset - parent.limit) : 0;
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/basic/basic_model.js:1303]: same here

**Comment by msh-odoo** [comment]
@aab-odoo Done changes, Thanks.

**Comment by SimonGenin** [comment]
@robodoo r+

**Pull request closed without merging**